src/cid_collections/small_cid_vec.rs

```diff
@@ -11,31 +12,31 @@ use crate::blocks::TipsetKey;
 /// There are typically MANY small, immutable collections of CIDs in, e.g [`TipsetKey`]s.
 ///
 /// Save space on those by:
-/// - Using a boxed slice to save on vector overallocation.
-///   (In the worst case, this uses half the memory)
 /// - Using [`SmallCid`]s
 ///   (In the median case, this uses 40 B over 96 B per CID)
 ///
 /// This may be expanded to have [`smallvec`](https://docs.rs/smallvec/1.11.0/smallvec/index.html)-style indirection
 /// to save more on heap allocations.
-#[derive(Default, Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, PartialOrd, Ord)]
-#[serde(transparent)] // treat the named field as anonymous, so we serialize equivalent to Vec<Cid>
-pub struct FrozenCidVec {
-    inner: Box<[SmallCid]>,
-}
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, PartialOrd, Ord)]
+pub struct SmallCidNonEmptyVec(NonEmpty<SmallCid>);
 
-impl FrozenCidVec {
-    /// Returns true if the slice has a length of 0.
-    ///
-    /// See also [`is_empty`](https://doc.rust-lang.org/std/primitive.slice.html#method.is_empty).
-    pub fn is_empty(&self) -> bool {
-        self.inner.is_empty()
-    }
+impl SmallCidNonEmptyVec {
     /// Returns `true` if the slice contains an element with the given value.
     ///
     /// See also [`contains`](https://doc.rust-lang.org/std/primitive.slice.html#method.contains).
     pub fn contains(&self, cid: Cid) -> bool {
-        self.inner.contains(&SmallCid::from(cid))
+        self.0.contains(&SmallCid::from(cid))
+    }
+
+    /// Returns a non-empty collection of `CID`
+    pub fn into_cids(self) -> NonEmpty<Cid> {
+        self.0.map(From::from)
+    }
+}
+
+impl Default for SmallCidNonEmptyVec {
+    fn default() -> Self {
+        Self(nonempty![Cid::default().into()])
     }
 }
```

Comment:
This does not seem right. Can you explain it to me?

Reply:
Yeah, I guess the idea here is that `nonempty!` is not actually empty. So it should not have a `Default` trait implemented, otherwise what's the point of it?

Reply:
Dropped `Default` derives for non test targets